### ISSUE TITLE
CLAP Note Expressions in BWS 4.2

### ISSUE THREAD

**Issue by baconpaul** [free-audio/interop-tracker#2]
In Multitrack Studio I get the first value of a note expression when a note is created.

In BWS the first note expression doesn't come until sample 0 of the subsequent block

So that is

clap::process
    - midi create voice mid-block

clap::proces
    - first clap note event

so as a result i can't initialize my note events.

**Comment by baconpaul** [comment]
Also I think understanding how BWS maps to the note expressions would help

For instance brightness maps to timbre. What maps to expression?

And is brightness bipolar or unipolar? the docs say // 0..1 at the start of the block but bw sends me -1 for brightness happiy

**Comment by baconpaul** [comment]
Similarly it seems BWS sends me pan messages in the range -1, 1 not in the range 0,1

**Comment by baconpaul** [comment]
that's pan note expressions I mean

**Comment by abique** [comment]
> Similarly it seems BWS sends me pan messages in the range -1, 1 not in the range 0,1

This bug is addressed for bitwig 4.2.1

**Comment by abique** [comment]
Timbre maps to brightness yes.

**Comment by abique** [comment]
I confirm the note expression bug:
```
CLAP: NoteOn, block: 13876, time: 200, port: 0, channel: 0, key: 41, vel: 0.787402
CLAP: NoteExpression, block: 13877, time: 0, port: 0, channel: 0, key: 41, exp_id: 5, value: 0.696
CLAP: NoteOff, block: 13894, time: 0, port: 0, channel: 0, key: 41, vel: 0.503937
```

**Comment by baconpaul** [comment]
> Timbre maps to brightness yes.

Great. And will I get 0..1 or -1..1 from BWS? The clap header makes me think 0..1 but the BW UI shows -1..1

**Comment by abique** [comment]
0..1 for the brightness with 4.2.1

**Comment by abique** [comment]
@baconpaul can you confirm the fix in 4.2.1?

**Comment by abique** [comment]
I close the issue, please re-open if necessary.
Thank :)

**Comment by baconpaul** [comment]
Yea sorry I will look this week but please close! Apologies got held up with some non surge work

**Comment by abique** [comment]
Be aware that there is a crash with pitch note expression in this build, already fixed for 4.2.3
